clang/test/CodeGenSYCL/invoke-function-addrspace.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_cc1 -fsycl -sycl-std=2020 -fsycl-is-device -fsycl-allow-func-ptr -internal-isystem %S/Inputs -disable-llvm-passes -triple spir64-unknown-unknown-sycldevice -emit-llvm -o - %s | FileCheck %s
+
+// Test that the type of function object invoked from the kernel has
+// the right address space.
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+// CHECK-DAG: define dso_local spir_func i32 @{{.*}}bar10{{.*}}()
```

Comment:
Is there a reason you used CHECK-DAG instead of CHECK? DAG is to not preserve order right?

Reply:
Yes, I used CHECK-DAG for the out of order checks.   I have now changed the test to use CHECK instead and have reordered the code.